lib/Yield/HexDump.cpp

```diff
@@ -0,0 +1,280 @@
+
+    CurrentAddress = Address;
+
+    std::optional<llvm::ArrayRef<uint8_t>>
+      SegmentBinary = BinaryView.getByAddress(Address, Size);
+
+    const size_t BytesInLine = 16;
+
+    size_t Counter = 0;
+
+    std::set<ptml::Tag> PreviousAddresses;
+    bool SpanIsOpen = false;
+    if (SegmentBinary.has_value()) {
+      SmallString<16> PrintableChars;
+      for (size_t Index = 0; Index < SegmentBinary->size(); ++Index) {
+        boost::icl::discrete_interval<MetaAddress>
+          CurrentInterval = (Current != End) ?
+                              Current->first :
+                              boost::icl::discrete_interval<MetaAddress>{
+                                MetaAddress{}, MetaAddress{}
+                              };
+        const bool LineBegins = Index % BytesInLine == 0;
+        if (LineBegins) {
+          // Print address of first byte in line
+          Output << FormatNumber(CurrentAddress.address()) << "  ";
+        }
+
+        // Open tags if this is beginning of next interval or line begins
+        const bool IsIntervalValid = CurrentInterval.lower().isValid()
+                                     and CurrentInterval.upper().isValid();
+        const bool AfterStart = CurrentInterval.lower() <= CurrentAddress;
+        const bool BeforeEnd = CurrentAddress < CurrentInterval.upper();
+        const bool IsInsideInterval = IsIntervalValid and AfterStart
+                                      and BeforeEnd;
+        const bool AtStart = CurrentInterval.lower() == CurrentAddress;
+        const bool IsStartOfInterval = IsIntervalValid and AtStart;
+        if (IsStartOfInterval or (LineBegins and IsInsideInterval)) {
+          for (const AddressTag &Tag : Current->second) {
+            Output << Tag.PTMLTag.open();
+          }
+        }
+
+        // Format number and put it to the output
+        llvm::ArrayRef<uint8_t> &S = *SegmentBinary;
+        const uint64_t &B = S[Index];
+
+        Output << FormatNumber(B, 2);
+        ++Counter;
+
+        if (std::isprint(B)) {
+          PrintableChars += B;
+        } else {
+          PrintableChars += '.';
+        }
+
+        MetaAddress NextAddress = CurrentAddress + 1;
+
+        const bool EndOfLine = Counter == BytesInLine;
+        const bool EndOfInterval = CurrentAddress < CurrentInterval.upper()
+                                   and NextAddress >= CurrentInterval.upper();
+
+        const bool EndOfSegment = Index + 1 == SegmentBinary->size();
+        if (IsInsideInterval and (EndOfInterval or EndOfLine)) {
+          for (const AddressTag &Tag : Current->second) {
+            Output << Tag.PTMLTag.close();
+          }
+        }
+
+        if (Counter == BytesInLine) {
+          // Output ASCII representation at the end of the line
+          std::string Temp;
+          raw_string_ostream Printable(Temp);
+          printHTMLEscaped(PrintableChars, Printable);
+          Output << "   | " << Printable.str() << " |\n";
```

Comment:
I never noticed `raw_string_ostream` has `.str`, I always ensure `raw_string_ostream` is destructed and then use the backing `std::string`. Nice!